src/clr-angular/utils/drag-and-drop/providers/drag-event-listener.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable, Renderer2} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+
+import {ClrDragDataTransfer, ClrDragEvent, ClrDragEventType} from "../interfaces/drag-event";
+import {ClrDragAndDropEventBus} from "./drag-and-drop-event-bus";
+
+@Injectable()
+export class ClrDragEventListener {
+    private draggabelEl: Node;
+
+    private listeners: [() => void];
+
+    private dragStart: Subject<ClrDragEvent> = new Subject<ClrDragEvent>();
+    private dragMove: Subject<ClrDragEvent> = new Subject<ClrDragEvent>();
+    private dragEnd: Subject<ClrDragEvent> = new Subject<ClrDragEvent>();
+
+    private hasDragStarted: boolean = false;
+
+    get dragStarted(): Observable<ClrDragEvent> {
+        return this.dragStart.asObservable();
+    }
+
+    get dragMoved(): Observable<ClrDragEvent> {
+        return this.dragMove.asObservable();
+    }
+
+    get dragEnded(): Observable<ClrDragEvent> {
+        return this.dragEnd.asObservable();
+    }
+
+    constructor(private renderer: Renderer2, private eventBus: ClrDragAndDropEventBus) {}
+
+    public dragDataTransfer?: ClrDragDataTransfer;
+
+    public attachDragListeners(draggableEl: Node) {
+        this.draggabelEl = draggableEl;
+        this.listeners = [
+            this.customDragEvent(this.draggabelEl, "mousedown", "mousemove", "mouseup"),
+            this.customDragEvent(this.draggabelEl, "touchstart", "touchmove", "touchend")
+        ];
+    }
+
+    public detachDragListeners() {
+        if (this.listeners) {
+            this.listeners.map(event => event());
+        }
+    }
+
+    private customDragEvent(element: Node, startOnEvent: string, moveOnEvent: string, endOnEvent: string): () => void {
+        let moveListener: () => void;
+        let endListener: () => void;
+
+        return this.renderer.listen(element, startOnEvent, () => {
+            moveListener = this.renderer.listen("document", moveOnEvent, (moveEvent: MouseEvent|TouchEvent) => {
+                moveEvent.preventDefault();
+                moveEvent.stopPropagation();
```

Comment:
We mentioned this before, we can never stop propagation. I don't remember why you added it here, but it's simply something we cannot have at all in a library.

Reply:
I think he mentioned something about nested draggables. Not sure though. But yes, we can't have `stopPropagation`.